src/main/rx/expresslrs_telemetry.c

```diff
@@ -248,17 +242,22 @@ void receiveMspData(const uint8_t packageIndex, const volatile uint8_t* receiveD
         return;
     }
 
-    if (packageIndex == mspCurrentPackage) {
-        for (uint8_t i = 0; i < mspBytesPerCall; i++) {
-            mspData[mspCurrentOffset++] = *(receiveData + i);
-        }
-
+    bool acceptData = false;
+    if (packageIndex == 0 && mspCurrentPackage > 1) {
+        // PackageIndex 0 (the final packet) can also contain data
+        acceptData = true;
+        finishedData = true;
+    } else if (packageIndex == mspCurrentPackage) {
+        acceptData = true;
         mspCurrentPackage++;
-        mspConfirm = !mspConfirm;
-        return;
     }
 
-    return;
+    if (acceptData) {
```

Comment:
Should we also be checking that `receiveData` isn't NULL here?

Reply:
it should never be NULL, we're processing the packet after the receive done IRQ which should contain full 8 bytes of data and a crc check. Shouldn't hurt to add the check

Reply:
added a `NULL` check